core/src/main/java/org/elasticsearch/rest/action/bulk/RestBulkAction.java

```diff
@@ -97,49 +94,19 @@ public RestResponse buildResponse(BulkResponse response, XContentBuilder builder
                 for (BulkItemResponse itemResponse : response) {
                     builder.startObject();
                     builder.startObject(itemResponse.getOpType());
-                    builder.field(Fields._INDEX, itemResponse.getIndex());
-                    builder.field(Fields._TYPE, itemResponse.getType());
-                    builder.field(Fields._ID, itemResponse.getId());
-                    long version = itemResponse.getVersion();
-                    if (version != -1) {
-                        builder.field(Fields._VERSION, itemResponse.getVersion());
-                    }
                     if (itemResponse.isFailed()) {
+                        builder.field(Fields._INDEX, itemResponse.getIndex());
+                        builder.field(Fields._TYPE, itemResponse.getType());
+                        builder.field(Fields._ID, itemResponse.getId());
```

Comment:
maybe we could go one step further and implement StatusToXContent on BulkItemResponse :-)